### PR TITLE
Divine exo upgrades

### LINKED ISSUE #337: Comprehensive Suggestions for Exo Armor
I put quite a bit of thought into this, and gathered them all into one place as requested. 
The focus of this is on expanding existing concepts already implemented, with the goal of three good component choices which aren't cross mod.

**Armor Plates**
* [x] Gilded Iron Plate: Weaker combo of Iron and Gold plates.
  * [x] Code
  * [x] Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang
* [x] Zinc Armor Plate: Ablative! Pops off when I would get below 5 hearts, and prevents up to 5 hearts of damage.
  * [x] Code
  * [x] Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang

**Helmet Slot**
* Conical Hat: Deflects falling items, and breaks them if they would become solid over your head, Great for people who keep digging up.
* Finned Helmet (I just watched the rocketeer): makes me go faster? or jerk in the direction I suddenly twitch the mouse?
* [x] Frequency Shifter: This odd acoustic invention Changes the frequencies you are hearing and speaking on. As a result you can trade with cats and dogs in their preferred currencies, and you don’t hear chat messages except from others wearing the shifter.
  * [x] Code
  * [x] Item Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang

**Mask Slot**
* [x] Rebreather: Restores breath bubbles at a steam cost.
  * [x] Code
  * [x] Item Texture
  * [x] Armor Texture
  * [x] Recipe
  * [x] Lang
* [x] Dragon’s roar: this Emergency Steam Vent outward over the mouth provides the ability to damage a whole group of enemies.
  * [x] Code
  * [x] Item Texture
  * [x] Model
    * Copper robotic dragon head mask
  * [x] Recipe
  * [x] Lang

**Arm Slot**
* [x] Piston Push: While not as satisfying as the powerfist, this simple brass piston allows me to move blocks away from me as if I was a piston. Surprisingly useful in exploring for caves while mining.
  * [x] Code
  * [x] Item Texture
  * [x] Armor Texture
  * [x] Recipe
  * [x] Lang
* Smasher Punch: Mounting a rock smasher on my arm was a brilliant idea, now I don’t need to spend as much time processing, as ores I break are already smashed into easily blendable components.

**Hip Slot**
* Recharger: Drains your Steam into steamholding items, greatly expanding the field time for steam powered tools.
* [x] Reloading Holsters: Reloads your pistols while you aren’t holding them.
  * [x] Code
  * [x] Item Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang
* Secondary Steam Tanks: Adds basic steam storage to your pants!

**Leg Slot**
* Gyroscopes: Keeping me steady, and removing nausea and 

**Heel Slot**
* [x] Anchor Heels: Lead Plates help hold me down, and rapidly sink me to the bottom of the ocean, as well as resist knockback.
  * [x] Code
  * [x] Item Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang
* Snowshoes: Walk up snow and ice blocks as if they were stairs. Great for if you’ve built your lab on a mountaintop.

**FootSlot**
* [x] Hydrophobic Coatings: Zinc and Slime exert water repulsive properties when heated, this should allow me to coast (as if sliding on ice) over the surface of water.
  * [x] Code
  * [x] Item Texture
  * [x] Armor texture
  * [x] Recipe
  * [x] Lang
* [x] Pyrophobic Coatings: Copper and Magma Creme, as above, only on lava and a lot more expensive.
  * [x] Code
  * [x] Item Texture
  * [x] Armor texture
  * [x] Recipe
  * [x] Lang

### PR DESCRIPTION
**Pull request by elifoster**
This PR handles everything in #337.

@Lance5057 @xbony2 @Strikingwolf If y'all could perform a code review that'd be really neat.

Close #337.

By merging this, we'll be significantly closer to 0.29!

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -2,3 +2,4 @@
 	path = src/main/resources
 	url = https://github.com/Esteemed-Innovation/Flaxbeards-Steam-Power-Resources.git
 	ignore = dirty
+	branch = divine-exo-upgrades
```
Comment (elifoster) [inline .gitmodules]: Oh, yikes.

**Comment by elifoster** [comment]
That travis failure is just because the commit SHA is outdated. I'd prefer to update that on master, so ignore the travis failure. It compiles.

File: src/main/java/flaxbeard/steamcraft/Config.java [hunk 2]
```diff
@@ -327,6 +368,7 @@ public static void load(FMLPreInitializationEvent event) {
         enableSadistPlate = config.get("Exosuit Plates", "Enable sadist plate", true).getBoolean(true);
         enableVibrantPlate = config.get("Exosuit Plates", "Enable vibrant plate", true).getBoolean(true);
         enableEnderiumPlate = config.get("Exosuit Plates", "Enable enderium plate", true).getBoolean(true);
+        enableGildedIronPlate = config.get("Exosuit Plates", "Enable gilded iron palte", true).getBoolean(true);
```
Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/Config.java]: That's some nice spelling there.

File: src/main/java/flaxbeard/steamcraft/SteamcraftBook.java [hunk 3]
```diff
@@ -433,10 +489,49 @@ public static void registerExosuit() {
...
+                    SteamcraftRegistry.addResearch(
+                      "research.AnchorHeels.name",
+                      "!research.ExoHeel.name",
+                      new BookPageItem("research.AnchorHeels.name",
+                        "research.AnchorHeelsIron.0", true,
+                        new ItemStack(SteamcraftItems.anchorHeels)),
+                      new BookPageCrafting("", "anchorHeels")
+                    );
+                }
+            }
```
Comment (Strikingwolf) [inline src/main/java/flaxbeard/steamcraft/SteamcraftBook.java]: There are essentially the same, I would have a boolean local variable defined as `Config.enableLeadPlate && OreDictionary.getOres("ingotLead").size() > 0 && !Config.enableAnchorAnvilRecipe` and then use a ternary to get whether it should be "research.AnchorHeelsIron.0" or "AnchorHeelsLead", you could even just insert the Iron/Lead part and have "research.AnchorHeels<insert>.0" common

File: src/main/java/flaxbeard/steamcraft/SteamcraftBook.java [hunk 4]
```diff
@@ -433,10 +489,49 @@ public static void registerExosuit() {
...
+                );
+            }
+            if (Config.enablePyrophobic) {
+                SteamcraftRegistry.addResearch(
+                  "research.Pyrophobic.name", "!research.ExoFoot.name",
+                  new BookPageItem("research.Pyrophobic.name", "research.Pyrophobic.0", true,
+                    new ItemStack(SteamcraftItems.coatingsPyrophobic)),
+                  new BookPageCrafting("", "pyrophobic")
+                );
+            }
```
Comment (Strikingwolf) [inline src/main/java/flaxbeard/steamcraft/SteamcraftBook.java]: This applies to the entire deal with Config and research, is there a way you could create a more general function for this? There seems to be a lot of repeated code. Possibly pass in a boolean array and if it passes that register a string and have another argument for the !

File: src/main/java/flaxbeard/steamcraft/handler/SteamcraftEventHandler.java [hunk 5]
```diff
@@ -819,6 +837,110 @@ public void handleFirePunch(LivingAttackEvent event) {
...
+                        world.playSoundEffect((double) x + 0.5D, (double) y + 0.5D, (double) z +
+                          0.5D, "tile.piston.out", 0.5F, world.rand.nextFloat() * 0.25F + 0.6F);
+                    }
+                }
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public void hearMeRoar(LivingAttackEvent event) {
```
Comment (xbony2) [inline src/main/java/flaxbeard/steamcraft/handler/SteamcraftEventHandler.java]: katy perry

**Pull request merged**